Index: core/portlet/ecm/ext/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManageVersionsActionComponent.java
===================================================================
--- core/portlet/ecm/ext/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManageVersionsActionComponent.java	(revision 64495)
+++ core/portlet/ecm/ext/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManageVersionsActionComponent.java	(working copy)
@@ -26,6 +26,7 @@
 import org.exoplatform.ecm.webui.component.explorer.control.filter.CanSetPropertyFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotRootNodeFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotLockedFilter;
+import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotWebContentChildNodeFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.listener.UIActionBarActionListener;
 import org.exoplatform.ecm.webui.component.explorer.versions.UIActivateVersion;
 import org.exoplatform.ecm.webui.component.explorer.versions.UIVersionInfo;
@@ -51,7 +52,9 @@
  )
 public class ManageVersionsActionComponent extends UIComponent {
 
-  private static final List<UIExtensionFilter> FILTERS = Arrays.asList(new UIExtensionFilter[]{new IsNotRootNodeFilter("UIActionBar.msg.cannot-enable-version-rootnode"), new CanSetPropertyFilter(), new IsNotLockedFilter(), new CanEnableVersionFilter()});
+  private static final List<UIExtensionFilter> FILTERS = Arrays.asList(new UIExtensionFilter[] {
+      new IsNotRootNodeFilter("UIActionBar.msg.cannot-enable-version-rootnode"),
+      new CanSetPropertyFilter(), new IsNotLockedFilter(), new CanEnableVersionFilter(), new IsNotWebContentChildNodeFilter()});
 
   @UIExtensionFilters
   public List<UIExtensionFilter> getFilters() {
Index: core/portlet/ecm/ext/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManagePublicationsActionComponent.java
===================================================================
--- core/portlet/ecm/ext/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManagePublicationsActionComponent.java	(revision 64495)
+++ core/portlet/ecm/ext/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/action/ManagePublicationsActionComponent.java	(working copy)
@@ -30,6 +30,7 @@
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsDocumentFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotRootNodeFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotLockedFilter;
+import org.exoplatform.ecm.webui.component.explorer.control.filter.IsNotWebContentChildNodeFilter;
 import org.exoplatform.ecm.webui.component.explorer.control.listener.UIActionBarActionListener;
 import org.exoplatform.ecm.webui.component.explorer.popup.admin.UIActivePublication;
 import org.exoplatform.ecm.webui.component.explorer.popup.admin.UIPublicationManager;
@@ -61,7 +62,12 @@
  )
 public class ManagePublicationsActionComponent extends UIComponent {
 
-  private static final List<UIExtensionFilter> FILTERS = Arrays.asList(new UIExtensionFilter[]{new HasPublicationLifecycleFilter(), new IsDocumentFilter("UIActionBar.msg.manage-publication.not-supported-nodetype"), new IsNotRootNodeFilter("UIActionBar.msg.cannot-enable-publication-rootnode"), new CanSetPropertyFilter("UIActionBar.msg.access-denied"), new CanRemoveNodeFilter(), new IsNotLockedFilter()});
+  private static final List<UIExtensionFilter> FILTERS = Arrays.asList(new UIExtensionFilter[] {
+      new HasPublicationLifecycleFilter(),
+      new IsDocumentFilter("UIActionBar.msg.manage-publication.not-supported-nodetype"),
+      new IsNotRootNodeFilter("UIActionBar.msg.cannot-enable-publication-rootnode"),
+      new CanSetPropertyFilter("UIActionBar.msg.access-denied"), new CanRemoveNodeFilter(),
+      new IsNotLockedFilter(), new IsNotWebContentChildNodeFilter()});
   
   @UIExtensionFilters
   public List<UIExtensionFilter> getFilters() {
Index: core/portlet/ecm/core/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/filter/IsNotWebContentChildNodeFilter.java
===================================================================
--- core/portlet/ecm/core/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/filter/IsNotWebContentChildNodeFilter.java	(revision 0)
+++ core/portlet/ecm/core/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/control/filter/IsNotWebContentChildNodeFilter.java	(revision 0)
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2003-2010 eXo Platform SAS.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see<http://www.gnu.org/licenses/>.
+ */
+package org.exoplatform.ecm.webui.component.explorer.control.filter;
+
+import java.util.Map;
+
+import javax.jcr.Node;
+
+import org.exoplatform.ecm.webui.utils.Utils;
+import org.exoplatform.services.jcr.impl.core.NodeImpl;
+import org.exoplatform.webui.ext.filter.UIExtensionFilter;
+import org.exoplatform.webui.ext.filter.UIExtensionFilterType;
+
+/**
+ * Created by The eXo Platform SAS
+ * Author : Dang Viet Ha
+ *          ha.dangviet@exoplatform.com
+ * Dec 28, 2010  
+ */
+public class IsNotWebContentChildNodeFilter implements UIExtensionFilter {
+
+  public boolean accept(Map<String, Object> context) throws Exception {
+    Node currentNode = (Node) context.get(Node.class.getName());
+    Node parrentNode = currentNode.getParent();
+    while (!((NodeImpl) parrentNode).isRoot()) {
+      if (parrentNode.isNodeType(Utils.EXO_WEBCONTENT)) {
+        return false;
+      }
+      parrentNode = parrentNode.getParent();
+    }
+    return true;
+  }
+
+  public UIExtensionFilterType getType() {
+    return UIExtensionFilterType.MANDATORY;
+  }
+  
+  public void onDeny(Map<String, Object> context) throws Exception {
+  }
+
+}
Index: core/webui/dms/src/main/java/org/exoplatform/ecm/webui/utils/Utils.java
===================================================================
--- core/webui/dms/src/main/java/org/exoplatform/ecm/webui/utils/Utils.java	(revision 64495)
+++ core/webui/dms/src/main/java/org/exoplatform/ecm/webui/utils/Utils.java	(working copy)
@@ -149,6 +149,7 @@
   final static public String EXO_HIDDENABLE = "exo:hiddenable";
   final static public String EXO_ACCESSPERMISSION = "exo:accessPermissions";
 
+  final static public String EXO_WEBCONTENT = "exo:webContent";
   final static public String EXO_MUSICFOLDER = "exo:musicFolder";
   final static public String EXO_VIDEOFOLDER = "exo:videoFolder";
   final static public String EXO_PICTUREFOLDER = "exo:pictureFolder";
