Index: core/portlet/ecm/core/web/src/main/webapp/javascript/eXo/ecm/UIListView.js
===================================================================
--- core/portlet/ecm/core/web/src/main/webapp/javascript/eXo/ecm/UIListView.js	(revision 71330)
+++ core/portlet/ecm/core/web/src/main/webapp/javascript/eXo/ecm/UIListView.js	(working copy)
@@ -89,13 +89,15 @@
 	};
 	
 	ListView.prototype.setScroll = function(evt){
-	  eXo.ecm.UIListView.object = this;
-    var element = eXo.ecm.UIListView.object ;
-	  var pos = eXo.core.Browser.findMouseYInPage(evt) - eXo.core.Browser.findPosY(element);
-	  if(element.offsetHeight - pos < 200){
-	    element.scrollTop = element.scrollTop + 5;  
-	  } else if(element.scrollTop > 0 && pos < 40) {
-	    element.scrollTop = element.scrollTop - 5;  
+	  if(Self.enableDragDrop) {
+	    eXo.ecm.UIListView.object = this;
+        var element = eXo.ecm.UIListView.object ;
+	    var pos = eXo.core.Browser.findMouseYInPage(evt) - eXo.core.Browser.findPosY(element);
+	    if(element.offsetHeight - pos < 200){
+	      element.scrollTop = element.scrollTop + 5;  
+	    } else if(element.scrollTop > 0 && pos < 40) {
+	      element.scrollTop = element.scrollTop - 5;  
+	    }
 	  }
 	};
 	
