Index: core/portlet/ecm/core/main/src/main/java/org/exoplatform/ecm/webui/component/admin/repository/UIWorkspaceWizard.java
===================================================================
--- core/portlet/ecm/core/main/src/main/java/org/exoplatform/ecm/webui/component/admin/repository/UIWorkspaceWizard.java	(revision 48461)
+++ core/portlet/ecm/core/main/src/main/java/org/exoplatform/ecm/webui/component/admin/repository/UIWorkspaceWizard.java	(working copy)
@@ -339,6 +339,7 @@
       if (!listType.contains(PermissionType.ADD_NODE)) rootNode.removePermission("any", PermissionType.ADD_NODE);
       if (!listType.contains(PermissionType.SET_PROPERTY)) rootNode.removePermission("any", PermissionType.SET_PROPERTY);
       if (!listType.contains(PermissionType.REMOVE)) rootNode.removePermission("any", PermissionType.REMOVE);
+      if (!listType.contains(PermissionType.READ)) rootNode.removePermission("any", PermissionType.READ); 
     }
     rootNode.save();
   }
