Index: core/component/cms/src/main/java/org/exoplatform/services/cms/webdav/WebDavServiceImpl.java
===================================================================
--- core/component/cms/src/main/java/org/exoplatform/services/cms/webdav/WebDavServiceImpl.java	(revision 63651)
+++ core/component/cms/src/main/java/org/exoplatform/services/cms/webdav/WebDavServiceImpl.java	(working copy)
@@ -183,6 +183,7 @@
   public Response get(@URIParam("repoName") String repoName,
                       @URIParam("repoPath") String repoPath,
                       @HeaderParam(WebDavHeaders.RANGE) String rangeHeader,
+                      @HeaderParam(WebDavHeaders.IF_MODIFIED_SINCE) String ifModifiedSince,
                       @QueryParam("version") String version,
                       @ContextParam(ResourceDispatcher.CONTEXT_PARAM_BASE_URI) String baseURI) {
 
@@ -197,7 +198,7 @@
       log.warn("Cannot find the item at " + repoName + "/" + repoPath, e);
       return Response.Builder.serverError().build();
     }
-    return super.get(repoName, repoPath, rangeHeader, version, baseURI);
+    return super.get(repoName, repoPath, rangeHeader, ifModifiedSince, version, baseURI);
   }
 
   @HTTPMethod(WebDavMethods.HEAD)
