Index: core/webui/dms/src/main/java/org/exoplatform/ecm/webui/form/UIDialogForm.java
===================================================================
--- core/webui/dms/src/main/java/org/exoplatform/ecm/webui/form/UIDialogForm.java	(revision 49285)
+++ core/webui/dms/src/main/java/org/exoplatform/ecm/webui/form/UIDialogForm.java	(working copy)
@@ -319,7 +319,10 @@
           if(childNode.getProperty(propertyName).getDefinition().isMultiple()) {
             Value[] values = childNode.getProperty(propertyName).getValues();
             for(Value value : values) {
-              uiDateTime.setCalendar(value.getDate());
+              if (uiDateTime.getDefaultValue() == null) {
+                uiDateTime.setCalendar(value.getDate());
+                uiDateTime.setDefaultValue(uiDateTime.getValue());
+              }
             }
           } else {
             uiDateTime.setCalendar(childNode.getProperty(propertyName).getValue().getDate());
