Index: core/portlet/ecm/core/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/UIJCRExplorer.java
===================================================================
--- core/portlet/ecm/core/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/UIJCRExplorer.java	(revision 50642)
+++ core/portlet/ecm/core/main/src/main/java/org/exoplatform/ecm/webui/component/explorer/UIJCRExplorer.java	(working copy)
@@ -687,13 +687,17 @@
     if (Preference.SORT_BY_NODENAME.equals(preferences_.getSortType())) {
       Collections.sort(childrenList, new NodeNameComparator(preferences_.getOrder())) ;
     } else if (Preference.SORT_BY_NODETYPE.equals(preferences_.getSortType())) {
-      Collections.sort(childrenList, new TypeNodeComparator(preferences_.getOrder())) ;
+        Collections.sort(childrenList, new TypeNodeComparator(preferences_.getOrder())) ;
     } else if (Preference.SORT_BY_VERSIONABLE.equals(preferences_.getSortType())) {
-      Collections.sort(childrenList, new StringComparator(preferences_.getOrder(), Preference.SORT_BY_VERSIONABLE));
+        Collections.sort(childrenList, new StringComparator(preferences_.getOrder(), Preference.SORT_BY_VERSIONABLE));
     } else if (Preference.SORT_BY_AUDITING.equals(preferences_.getSortType())) {
-      Collections.sort(childrenList, new StringComparator(preferences_.getOrder(), Preference.SORT_BY_AUDITING));
+        Collections.sort(childrenList, new StringComparator(preferences_.getOrder(), Preference.SORT_BY_AUDITING));
+    } else if (Preference.SORT_BY_CREATED_DATE.equals(preferences_.getSortType())) {
+        Collections.sort(childrenList, new PropertyValueComparator(Utils.EXO_CREATED_DATE, preferences_.getOrder())); 
+    } else if (Preference.SORT_BY_MODIFIED_DATE.equals(preferences_.getSortType())) {
+        Collections.sort(childrenList, new PropertyValueComparator(Utils.EXO_MODIFIED_DATE, preferences_.getOrder())); 
     } else {
-      Collections.sort(childrenList, new PropertyValueComparator(preferences_.getSortType(), preferences_.getOrder()));
+        Collections.sort(childrenList, new PropertyValueComparator(preferences_.getSortType(), preferences_.getOrder()));
     }
   }
   
@@ -823,4 +827,4 @@
       return path;
     }    
   }
-}
\ No newline at end of file
+}
